Modify S4423: Learn-As-You-Code Migration (#2097)
SonarSource/rspec: rules/S4423/rationale.adoc

Encryption algorithms are essential for protecting sensitive information and
ensuring secure communications in a variety of domains. They are used for
several important reasons:

* Confidentiality, privacy, and intellectual property protection
* Security during transmission or on storage devices
* Data integrity, general trust, and authentication

When selecting encryption algorithms, tools, or combinations, you should also
consider two things:

1. No encryption is unbreakable.

> 2. The strength of an encryption algorithm is usually measured by the effort required to crack it within a reasonable time frame.

Author: ```suggestion
2. The strength of an encryption algorithm is usually measured by the effort required to crack it within a reasonable time frame.

```

For these reasons, as soon as cryptography is included in a project, it is
important to choose encryption algorithms that are considered the strongest by
the cryptography community.

To provide communication security over a network, SSL and TLS are generally
used. However, it is important to note that the following protocols are all
considered weak by the cryptographic community, and are officially deprecated:

* SSL versions 1.0, 2.0 and 3.0.
* TLS versions 1.0 and 1.1.

> These two TLS versions are deprecated since
> https://datatracker.ietf.org/doc/html/rfc8996[March 2021].

Author: ```suggestion
When these unsecured protocols are used, it is best practice to expect a breach:
that a user or organization with malicious intent will perform mathematical
attacks on this data after obtaining it by other means.
```

> Encryption algorithms are essential for protecting sensitive information and

Reviewer: This file is the same (or very similar) as the file in the other PR that I already reviewed. It would be great if you could apply the same things here, so I don't have to write everything twice. For example, the dots at the end of the list entries.

Author: done

> Encryption algorithms are essential for protecting sensitive information and
> ensuring secure communications in a variety of domains. They are used for
> several important reasons:
> 
> * Confidentiality, privacy, and intellectual property protection.
> * Security during transmission or on storage devices.
> * Data integrity, general trust, and authentication.
> 
> When selecting encryption algorithms, tools, or combinations, you should also
> consider two things:
> 
> 1. No encryption is absolutely unbreakable.
> 2. The strength of an encryption is usually measured by the effort required to crack it within a reasonable time frame.
> 
> For these reasons, as soon as cryptography is included in a project, it is
> important to choose encryption algorithms that are considered the strongest by
> the cryptography community.

Author: ```suggestion
Encryption algorithms are essential for protecting sensitive information and
ensuring secure communications in a variety of domains. They are used for
several important reasons:

* Confidentiality, privacy, and intellectual property protection
* Security during transmission or on storage devices
* Data integrity, general trust, and authentication

When selecting encryption algorithms, tools, or combinations, you should also
consider two things:

1. No encryption is unbreakable.
2. The strength of an encryption algorithm is usually measured by the effort required to crack it within a reasonable time frame.
For these reasons, as soon as cryptography is included in a project, it is
important to choose encryption algorithms that are considered strong and secure
by the cryptography community.
```

To provide communication security over a network, SSL and TLS are generally
used. However, it is important to note that the following protocols are all
considered weak by the cryptographic community, and are officially deprecated:

> * SSL versions 1.0, 2.0 and 3.0.
> * TLS versions 1.0 and 1.1.

Author: ```suggestion
* SSL versions 1.0, 2.0 and 3.0
* TLS versions 1.0 and 1.1
```
